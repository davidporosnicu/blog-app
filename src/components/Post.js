import React, { Component } from "react";
import "../css/posts.css";

class Post extends Component {
  render() {
    const { title, body, tags } = this.props.post;
    return (
      <div className="blogSeparator">
        <div className="postTitle">
          <h2>{title}</h2>
        </div>
        <div className="postContent">
          <p>{body}</p>
          <ul className="ulStyle">
            {tags
              ? tags.map(tag => {
                  return <li className="tagStyle">{tag}</li>;
                })
              : null}
          </ul>
        </div>
      </div>
    );
  }
}

export default Post;
