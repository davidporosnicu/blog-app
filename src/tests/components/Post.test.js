import React from 'react';
import { shallow } from 'enzyme';
import Post from '../../components/Post';
import toJSON from 'enzyme-to-json';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });
describe('Post Component', () => {
  test('should render Post component correctly', () => {
    const post = {
      comments: [],
      entry: {}
    };
    const wrapper = shallow(<Post post={post} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
