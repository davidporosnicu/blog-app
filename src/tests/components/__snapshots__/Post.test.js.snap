// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post Component should render Post component correctly 1`] = `
<div
  className="blogSeparator"
>
  <div
    className="postTitle"
  >
    <h2 />
  </div>
  <div
    className="postContent"
  >
    <p />
    <ul
      className="ulStyle"
    />
    <Comments
      comments={Array []}
    />
  </div>
</div>
`;
